seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkWriter.java

```diff
@@ -0,0 +1,186 @@
+import org.influxdb.dto.Point;
+
+import java.io.IOException;
+import java.net.ConnectException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class InfluxDBSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private final Serializer serializer;
+    private InfluxDB influxDB;
+    private SinkConfig sinkConfig;
+    private final List<Point> batchList;
+
+    private ScheduledExecutorService scheduler;
+    private ScheduledFuture<?> scheduledFuture;
+    private volatile boolean initialize;
+    private volatile Exception flushException;
+    private final Integer batchIntervalMs;
+
+    public InfluxDBSinkWriter(Config pluginConfig,
+                              SeaTunnelRowType seaTunnelRowType) throws ConnectException {
+        this.sinkConfig = SinkConfig.loadConfig(pluginConfig);
+        this.batchIntervalMs = sinkConfig.getBatchIntervalMs();
+        this.serializer = new DefaultSerializer(
+                seaTunnelRowType, sinkConfig.getPrecision().getTimeUnit(), sinkConfig.getKeyTags(), sinkConfig.getKeyTime(), sinkConfig.getMeasurement());
+        this.batchList = new ArrayList<>();
+
+        connect();
+    }
+
+    @Override
+    public void write(SeaTunnelRow element) throws IOException {
+        Point record = serializer.serialize(element);
+        write(record);
+    }
+
+    @SneakyThrows
+    @Override
+    public Optional<Void> prepareCommit() {
+        // Flush to storage before snapshot state is performed
+        flush();
+        return super.prepareCommit();
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+            scheduler.shutdown();
+        }
+
+        flush();
+
+        if (influxDB != null) {
+            influxDB.close();
+            influxDB = null;
+        }
+    }
+
+    private void tryInit() throws IOException {
+        if (initialize) {
+            return;
+        }
+        initialize = true;
+        connect();
+        if (batchIntervalMs != null) {
+            scheduler = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("influxDB-sink-output-%s").build());
+            scheduledFuture = scheduler.scheduleAtFixedRate(
+                () -> {
+                    try {
+                        flush();
+                    } catch (IOException e) {
+                        flushException = e;
+                    }
+                },
+                    batchIntervalMs,
+                    batchIntervalMs,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public synchronized void write(Point record) throws IOException {
+        tryInit();
```

Comment:
Why not build the scheduler thread in the constructor of InfluxDBSinkWriter

Reply:
> Why not build the scheduler thread in the constructor of InfluxDBSinkWriter

thinks，i have fixed it. PTAL @ic4y